In [1]:
cd /mnt/data/fastq/rnaseq/

/mnt/data/fastq/rnaseq


In [2]:
ls


293/                        SRR629568_pass_2.fastq.gz
khmer_countgraph/           SRR629569_pass_1.fastq
khmer_k32/                  SRR629569_pass_1.fastq.gz
metadata.csv                SRR629569_pass_2.fastq
metadata.json               SRR629569_pass_2.fastq.gz
MG/                         SRR629570_pass_1.fastq
OS/                         SRR629570_pass_1.fastq.gz
sratoolkit.tar.gz           SRR629570_pass_2.fastq
SRR4098609_pass_1.fastq     SRR629570_pass_2.fastq.gz
SRR4098609_pass_1.fastq.gz  SRR629571_pass_1.fastq
SRR4098609_pass_2.fastq     SRR629571_pass_1.fastq.gz
SRR4098609_pass_2.fastq.gz  SRR629571_pass_2.fastq
SRR4098610_pass_1.fastq     SRR629571_pass_2.fastq.gz
SRR4098610_pass_1.fastq.gz  SRR629572_pass_1.fastq
SRR4098610_pass_2.fastq     SRR629572_pass_1.fastq.gz
SRR4098610_pass_2.fastq.gz  SRR629572_pass_2.fastq
SRR629557_pass_1.fastq      SRR629572_pass_2.fastq.gz
SRR629557_pass_1.fastq.gz   SRR629573_pass_1.fastq
SRR629557_pass_2.fastq      SRR629573_pass_1.fastq.gz
S

In [3]:
import pandas as pd

metadata = pd.read_csv('metadata.csv')
metadata.head()

LIBRARY_LAYOUT LIBRARY_NAME LIBRARY_SELECTION  LIBRARY_SOURCE  \
0  {'PAIRED': ''}          NaN       unspecified  TRANSCRIPTOMIC   
1  {'PAIRED': ''}          NaN       unspecified  TRANSCRIPTOMIC   
2  {'PAIRED': ''}          NaN       unspecified  TRANSCRIPTOMIC   
3  {'PAIRED': ''}          NaN       unspecified  TRANSCRIPTOMIC   
4  {'PAIRED': ''}          NaN       unspecified  TRANSCRIPTOMIC   

  LIBRARY_STRATEGY        acc cluster_name  is_public  load_done  \
0          RNA-Seq  SRR629557       public       True       True   
1          RNA-Seq  SRR629559       public       True       True   
2          RNA-Seq  SRR629561       public       True       True   
3          RNA-Seq  SRR629562       public       True       True   
4          RNA-Seq  SRR629563       public       True       True   

                    name  static_data_available  status  total_bases  \
0     A-431, replicate a                   True  public   5256416326   
1    A-431, replicates b                   True  public   6429905026   
2  U-251 MG, replicate a                   True  public   5791489076   
3  U-251 MG, replicate b                   True  public   2451533610   
4    U-2 OS, replicate a                   True  public   2069977426   

   total_spots  ver replicate                name_v2 cell_line  
0     26021863    2         a     A-431, replicate a     A-431  
1     31831213    2         b    A-431, replicates b     A-431  
2     28670738    1         a  U-251 MG, replicate a  U-251 MG  
3     12136305    1         b  U-251 MG, replicate b  U-251 MG  
4     10247413    2         a    U-2 OS, replicate a    U-2 OS

In [20]:
metadata.tail()

LIBRARY_LAYOUT LIBRARY_NAME LIBRARY_SELECTION  LIBRARY_SOURCE  \
18  {'PAIRED': ''}          NaN       unspecified  TRANSCRIPTOMIC   
19  {'PAIRED': ''}          NaN       unspecified  TRANSCRIPTOMIC   
20  {'PAIRED': ''}          NaN       unspecified  TRANSCRIPTOMIC   
21  {'PAIRED': ''}     SHSY5Y_a       unspecified  TRANSCRIPTOMIC   
22  {'PAIRED': ''}     SHSY5Y_b       unspecified  TRANSCRIPTOMIC   

   LIBRARY_STRATEGY         acc cluster_name  is_public  load_done  \
18          RNA-Seq   SRR629581       public       True       True   
19          RNA-Seq   SRR629582       public       True       True   
20          RNA-Seq   SRR629583       public       True       True   
21          RNA-Seq  SRR4098609       public       True       True   
22          RNA-Seq  SRR4098610       public       True       True   

                                       name  static_data_available  status  \
18                        PC-3, replicate b                   True  public   
19                        RT-4, replicate a                   True  public   
20                        RT-4, replicate b                   True  public   
21  RNA-seq of cell line SHSY5Y replicate a                   True  public   
22  RNA-seq of cell line SHSY5Y replicate b                   True  public   

    total_bases  total_spots  ver replicate             name_v2 cell_line  
18   2414023422     11950611    1         b   PC-3, replicate b      PC-3  
19   4375683802     21661801    1         a   RT-4, replicate a      RT-4  
20   4499721700     22275850    1         b   RT-4, replicate b      RT-4  
21   7465997770     36960385    1         a  SHSY5Y replicate a    SHSY5Y  
22   3592995008     17787104    1         b  SHSY5Y replicate b    SHSY5Y

In [21]:
metadata.query('cell_line == "U-2 OS"')

LIBRARY_LAYOUT LIBRARY_NAME LIBRARY_SELECTION  LIBRARY_SOURCE  \
4  {'PAIRED': ''}          NaN       unspecified  TRANSCRIPTOMIC   

  LIBRARY_STRATEGY        acc cluster_name  is_public  load_done  \
4          RNA-Seq  SRR629563       public       True       True   

                  name  static_data_available  status  total_bases  \
4  U-2 OS, replicate a                   True  public   2069977426   

   total_spots  ver replicate              name_v2 cell_line  
4     10247413    2         a  U-2 OS, replicate a    U-2 OS

Note: HEK 293 didn't work because the cell line has a space so the filename has a space :(. Need to rerun using the sanitized cell line name. Should skip the ones that are already done since we don't use `--force`.

In [4]:
! load-into-counting.py --help


|| This is the script load-into-counting.py in khmer.
|| You are running khmer version 2.1.1
|| You are also using screed version 1.0
||
|| If you use this script in a publication, please cite EACH of the following:
||
||   * MR Crusoe et al., 2015. http://dx.doi.org/10.12688/f1000research.6924.1
||   * Q Zhang et al., http://dx.doi.org/10.1371/journal.pone.0101271
||   * A. Döring et al. http://dx.doi.org:80/10.1186/1471-2105-9-11
||
|| Please see http://khmer.readthedocs.io/en/latest/citations.html for details.

usage: load-into-counting.py [--version] [--info] [-h] [-k KSIZE]
                             [-U UNIQUE_KMERS] [--fp-rate FP_RATE]
                             [-M MAX_MEMORY_USAGE] [--small-count]
                             [-T THREADS] [-b] [-s FORMAT] [-f] [-q]
                             output_countgraph_filename
                             input_sequence_filename
                             [input_sequence_filename ...]

Build a k-mer countgraph from the given s

In [17]:
for cell_line, df in metadata.groupby('cell_line'):
    print(cell_line, df.shape,)
    read1 = df.acc + "_pass_1.fastq.gz"
    read2 = df.acc + "_pass_2.fastq.gz"
    reads = ' '.join(read1) + ' ' + ' '.join(read2)
    
    # Sanitize the cell line name
    cell_line = cell_line.replace(' ', '_')
    output_folder = f"/mnt/data/khmer_k32/{cell_line}"
    output_file = f'{output_folder}/khmer_countgraph'
#     print(output_file)
    ! mkdir -p $output_folder
    # ksize of 32 is the default
    ! load-into-counting.py --summary-info tsv --max-memory-usage 60G --ksize 32 --threads 16  $output_file $reads
    

A-431 (2, 18)

|| This is the script load-into-counting.py in khmer.
|| You are running khmer version 2.1.1
|| You are also using screed version 1.0
||
|| If you use this script in a publication, please cite EACH of the following:
||
||   * MR Crusoe et al., 2015. http://dx.doi.org/10.12688/f1000research.6924.1
||   * Q Zhang et al., http://dx.doi.org/10.1371/journal.pone.0101271
||   * A. Döring et al. http://dx.doi.org:80/10.1186/1471-2105-9-11
||
|| Please see http://khmer.readthedocs.io/en/latest/citations.html for details.


PARAMETERS:
 - kmer size =     32 		(-k)
 - n tables =      4 		(-N)
 - max tablesize = 1.5e+10 	(-x)
Estimated memory usage is 60.0 Gb (6e+10 bytes = 4 bytes x 1.5e+10 entries / 1 entries per byte)
--------
ERROR: Not enough free space on disk for saved graph files;
       Need at least 15.0 GB more.
       Table size: 15.0 GB
       Free space: 0.0 GB
NOTE: This can be overridden using the --force argument
A-549 (2, 18)

|| This is the script load-into-count

SHSY5Y (2, 18)

|| This is the script load-into-counting.py in khmer.
|| You are running khmer version 2.1.1
|| You are also using screed version 1.0
||
|| If you use this script in a publication, please cite EACH of the following:
||
||   * MR Crusoe et al., 2015. http://dx.doi.org/10.12688/f1000research.6924.1
||   * Q Zhang et al., http://dx.doi.org/10.1371/journal.pone.0101271
||   * A. Döring et al. http://dx.doi.org:80/10.1186/1471-2105-9-11
||
|| Please see http://khmer.readthedocs.io/en/latest/citations.html for details.


PARAMETERS:
 - kmer size =     32 		(-k)
 - n tables =      4 		(-N)
 - max tablesize = 1.5e+10 	(-x)
Estimated memory usage is 60.0 Gb (6e+10 bytes = 4 bytes x 1.5e+10 entries / 1 entries per byte)
--------
ERROR: Not enough free space on disk for saved graph files;
       Need at least 15.0 GB more.
       Table size: 15.0 GB
       Free space: 0.0 GB
NOTE: This can be overridden using the --force argument
U-2 OS (1, 18)
mkdir: cannot create directory ‘/mnt

In [ ]:
range(2)

In [ ]:
ls -lh /mnt/data/khmer_k32/*/*

In [6]:
import numpy as np

In [8]:
np.power(4., -16)

2.3283064365386963e-10

In [10]:
1e8*np.log(1-np.power(4., -16))

-0.02328306436809747

In [11]:
np.exp(1e8*np.log(1-np.power(4., -16)))

0.97698589473394

In [12]:
1- np.exp(1e8*np.log(1-np.power(4., -16)))

0.02301410526605996

In [14]:
desired_probability = 0.01
genome_size = 1e8

np.log(genome_size*(1-desired_probability)/desired_probability)

23.015800594086954

In [15]:
desired_probability = 0.001
genome_size = 1e8

np.log(genome_size*(1-desired_probability)/desired_probability)

25.32743552260092

In [16]:
desired_probability = 1e-6
genome_size = 1e8

np.log(genome_size*(1-desired_probability)/desired_probability)

32.23619030191614

In [19]:
! extract-paired-reads.py --help


|| This is the script extract-paired-reads.py in khmer.
|| You are running khmer version 2.1.1
|| You are also using screed version 1.0
||
|| If you use this script in a publication, please cite EACH of the following:
||
||   * MR Crusoe et al., 2015. http://dx.doi.org/10.12688/f1000research.6924.1
||
|| Please see http://khmer.readthedocs.io/en/latest/citations.html for details.

usage: extract-paired-reads.py [--version] [--info] [-h] [-d OUTPUT_DIR]
                               [-p filename] [-s filename] [-f]
                               [--gzip | --bzip]
                               [infile]

Take a mixture of reads and split into pairs and orphans.

positional arguments:
  infile

optional arguments:
  --version             show program's version number and exit
  --info                print citation information
  -h, --help            show this help message and exit
  -d OUTPUT_DIR, --output-dir OUTPUT_DIR
                        Output split reads to specified directory.